File: corporate-reporting/src/types/PartnerOrganisationRisk.ts
import { IEntity } from "./Entity";
import { IUser } from "./User";
import { IPartnerOrganisation } from "./PartnerOrganisation";
import { IPartnerOrganisationRiskMitigationAction } from "./PartnerOrganisationRiskMitigationAction";
import { IPartnerOrganisationRiskRiskType } from "./PartnerOrganisationRiskRiskType";
import { IReportingEntity, ReportingEntity } from "./ReportingEntity";
import { IPartnerOrganisationRiskUpdate } from "./PartnerOrganisationRiskUpdate";

export interface IPartnerOrganisationRisk extends IReportingEntity {
    RiskCode: string;
    PartnerOrganisationID: number;
    RiskOwnerUserID: number;
    BeisRiskOwnerUserID: number;
    RiskEventDescription: string;
    RiskCauseDescription: string;
    RiskImpactDescription: string;
    UnmitigatedRiskProbabilityID: number;
    UnmitigatedRiskImpactLevelID: number;
    TargetRiskProbabilityID: number;
    TargetRiskImpactLevelID: number;

    BEISUnmitigatedRiskProbabilityID: number;
    BEISUnmitigatedRiskImpactLevelID: number;
    BEISTargetRiskProbabilityID: number;
    BEISTargetRiskImpactLevelID: number;

    RiskAppetiteID: number;
    BeisRiskAppetiteID: number; // Obsolete? Replaced with calculated appetite based on risk types
    DepartmentalObjectiveID: number;
    RiskIsOngoing: boolean;
    RiskProximity: Date;
    PartnerOrganisation?: IPartnerOrganisation;
    RiskOwnerUser?: IUser;
    BeisRiskOwnerUser?: IUser;
    PartnerOrganisationRiskMitigationActions?: IPartnerOrganisationRiskMitigationAction[];
    PartnerOrganisationRiskRiskTypes?: IPartnerOrganisationRiskRiskType[];
    UnmitigatedRiskProbability?: IEntity;
    UnmitigatedRiskImpactLevel?: IEntity;
    TargetRiskProbability?: IEntity;
    TargetRiskImpactLevel?: IEntity;

    BEISUnmitigatedRiskProbability?: IEntity;
    BEISUnmitigatedRiskImpactLevel?: IEntity;
    BEISTargetRiskProbability?: IEntity;
    BEISTargetRiskImpactLevel?: IEntity;

    RiskAppetite?: IEntity;
    BeisRiskAppetite?: IEntity;

    PartnerOrganisationRiskUpdates?: IPartnerOrganisationRiskUpdate[];
}

export class PartnerOrganisationRisk extends ReportingEntity implements IPartnerOrganisationRisk {
    public RiskCode = '';
    public PartnerOrganisationID = null;
    public RiskOwnerUserID = null;
    public BeisRiskOwnerUserID = null;
    public RiskEventDescription = '';
    public RiskCauseDescription = '';
    public RiskImpactDescription = '';
    public UnmitigatedRiskProbabilityID = null;
    public UnmitigatedRiskImpactLevelID = null;
    public TargetRiskProbabilityID = null;
    public TargetRiskImpactLevelID = null;

    public BEISUnmitigatedRiskProbabilityID = null;
    public BEISUnmitigatedRiskImpactLevelID = null;
    public BEISTargetRiskProbabilityID = null;
    public BEISTargetRiskImpactLevelID = null;

    public RiskAppetiteID = null;
    public BeisRiskAppetiteID = null;
    public DepartmentalObjectiveID = null;
    public RiskIsOngoing = false;
    public RiskProximity = null;
    public PartnerOrganisationRiskRiskTypes = [];
    public Contributors = [];
}

File: ORB.Core/Models/PartnerOrganisationRisk.cs
using System;
using System.Collections.Generic;

namespace ORB.Core.Models
{
    public partial class PartnerOrganisationRisk : ReportingSubEntity
    {
        public PartnerOrganisationRisk()
        {
            PartnerOrganisationRiskMitigationActions = new HashSet<PartnerOrganisationRiskMitigationAction>();
            PartnerOrganisationRiskRiskTypes = new HashSet<PartnerOrganisationRiskRiskType>();
            PartnerOrganisationRiskUpdates = new HashSet<PartnerOrganisationRiskUpdate>();
        }

        public string RiskCode { get; set; }
        public int PartnerOrganisationID { get; set; }
        public int? RiskOwnerUserID { get; set; }
        public int? BeisRiskOwnerUserID { get; set; }
        public int? RagOptionID { get; set; }
        public string RiskEventDescription { get; set; }
        public string RiskCauseDescription { get; set; }
        public string RiskImpactDescription { get; set; }
        public int? UnmitigatedRiskProbabilityID { get; set; }
        public int? UnmitigatedRiskImpactLevelID { get; set; }
        public int? TargetRiskProbabilityID { get; set; }
        public int? TargetRiskImpactLevelID { get; set; }
        public int? BEISUnmitigatedRiskProbabilityID { get; set; }
        public int? BEISUnmitigatedRiskImpactLevelID { get; set; }
        public int? BEISTargetRiskProbabilityID { get; set; }
        public int? BEISTargetRiskImpactLevelID { get; set; }
        public int? RiskAppetiteID { get; set; }
        public int? BeisRiskAppetiteID { get; set; }
        public int? DepartmentalObjectiveID { get; set; }
        public DateTime? RiskProximity { get; set; }
        public bool? RiskIsOngoing { get; set; }

        public virtual RiskAppetite BeisRiskAppetite { get; set; }
        public virtual User BeisRiskOwnerUser { get; set; }
        public virtual RiskImpactLevel BEISTargetRiskImpactLevel { get; set; }
        public virtual RiskProbability BEISTargetRiskProbability { get; set; }
        public virtual RiskImpactLevel BEISUnmitigatedRiskImpactLevel { get; set; }
        public virtual RiskProbability BEISUnmitigatedRiskProbability { get; set; }
        public virtual DepartmentalObjective DepartmentalObjective { get; set; }
        public virtual PartnerOrganisation PartnerOrganisation { get; set; }
        public virtual RagOption RagOption { get; set; }
        public virtual RiskAppetite RiskAppetite { get; set; }
        public virtual User RiskOwnerUser { get; set; }
        public virtual RiskImpactLevel TargetRiskImpactLevel { get; set; }
        public virtual RiskProbability TargetRiskProbability { get; set; }
        public virtual RiskImpactLevel UnmitigatedRiskImpactLevel { get; set; }
        public virtual RiskProbability UnmitigatedRiskProbability { get; set; }
        public virtual ICollection<PartnerOrganisationRiskMitigationAction> PartnerOrganisationRiskMitigationActions { get; set; }
        public virtual ICollection<PartnerOrganisationRiskRiskType> PartnerOrganisationRiskRiskTypes { get; set; }
        public virtual ICollection<PartnerOrganisationRiskUpdate> PartnerOrganisationRiskUpdates { get; set; }
    }
}
